internal/pkg/k8smeta/k8smeta.go

```diff
@@ -49,20 +39,10 @@ func InitObjectMeta(cfg *config.Config, objectMeta *metav1.ObjectMeta, composeSe
 // FindFromObjectMeta finds a docker compose service from resource metadata.
 func FindFromObjectMeta(cfg *config.Config, objectMeta *metav1.ObjectMeta) (*config.Service, error) {
 	if composeServiceName, ok := objectMeta.Annotations[AnnotationName]; ok {
-		composeService := cfg.CanonicalComposeFile.Services[composeServiceName]
+		composeService := cfg.FindServiceByName(composeServiceName)
 		if composeService != nil {
 			return composeService, nil
 		}
 	}
-	i := strings.IndexByte(objectMeta.Name, '-')
```

Comment:
This piece of code broke when we allowed hyphens in the middle of escaped name, so removed it. This code didn't do much anyway other than highlight a potential issue with concurrent runs of `kube-compose` on one environment, or interference of cleanup tasks with `kube-compose`.